internal/export/dashboards.go

```diff
@@ -0,0 +1,117 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package export
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/elastic-package/internal/common"
+	"github.com/elastic/elastic-package/internal/kibana"
+	"github.com/elastic/elastic-package/internal/logger"
+	"github.com/elastic/elastic-package/internal/packages"
+)
+
+// Dashboards method exports selected dashboards with references objects. All Kibana objects are saved to local files
+// in appropriate directories.
+func Dashboards(kibanaClient *kibana.Client, dashboardsIDs []string) error {
+	packageRoot, err := packages.MustFindPackageRoot()
+	if err != nil {
+		return errors.Wrap(err, "locating package root failed")
+	}
+	logger.Debugf("Package root found: %s", packageRoot)
+
+	m, err := packages.ReadPackageManifest(filepath.Join(packageRoot, packages.PackageManifestFile))
+	if err != nil {
+		return errors.Wrapf(err, "reading package manifest failed (path: %s)", packageRoot)
+	}
+
+	objects, err := kibanaClient.Export(dashboardsIDs)
+	if err != nil {
+		return errors.Wrap(err, "exporting dashboards using Kibana client failed")
+	}
+
+	ctx := &transformationContext{
+		packageName: m.Name,
+	}
+
+	objects, err = applyTransformations(ctx, objects)
+	if err != nil {
+		return errors.Wrap(err, "can't transform Kibana objects")
+	}
+
+	err = saveObjectsToFiles(packageRoot, objects)
+	if err != nil {
+		return errors.Wrap(err, "can't save Kibana objects")
+	}
+	return nil
+}
+
+func applyTransformations(ctx *transformationContext, objects []common.MapStr) ([]common.MapStr, error) {
+	return transformObjects(ctx, objects,
+		filterUnsupportedTypes,
+		decodeObject,
+		stripObjectProperties,
+		standardizeObjectProperties,
+		standardizeObjectID)
+}
```

Comment:
I like what you have done here with the transformation functions. Most of them are specific to Kibana objects, so I'm wondering if we should define them under `internal/export/kibana` instead of just `internal/export`? I'm okay with doing this later when we have exports of non-Kibana assets.

Reply:
Hm.. maybe postpone it for later once we have a conflict between data sources. Right now there is Kibana only, probably when we have Elasticsearch we'll have to adjust it and maybe change the package API.